InvenTree/plugin/installer.py

```diff
@@ -0,0 +1,223 @@
+"""Install a plugin into the python virtual environment"""
+
+import logging
+import os
+import re
+import subprocess
+import sys
+
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.utils.translation import gettext_lazy as _
+
+logger = logging.getLogger('inventree')
+
+
+def pip_command(*args):
+    """Build and run a pip command using using the current python executable
+
+    returns: subprocess.check_output
+    throws: subprocess.CalledProcessError
+    """
+
+    python = sys.executable
+
+    command = [python, '-m', 'pip']
+    command.extend(args)
+
+    logger.info(f"running pip command: {' '.join(command)}")
+
+    # Debug information, might be useful for working out edge cases
+    logger.debug(f"base prefix: {sys.base_prefix}")
+    logger.debug(f"python path: {sys.path}")
+
+    return subprocess.check_output(
+        command,
+        cwd=settings.BASE_DIR.parent,
+        stderr=subprocess.STDOUT,
+    )
+
+
+def check_package_path(packagename):
+    """Determine the install path of a particular package
+
+    - If installed, return the installation path
+    - If not installed, return False
+    """
+
+    logger.info(f"check_package_path: {packagename}")
+
+    try:
+        result = pip_command('show', packagename)
+
+        output = result.decode('utf-8').split("\n")
+
+        for line in output:
+            # Check if line matches pattern "Location: ..."
+            match = re.match(r'^Location:\s+(.+)$', line.strip())
+
+            if match:
+                return match.group(1)
+
+    except subprocess.CalledProcessError as error:
+
+        output = error.output.decode('utf-8')
+        logger.error(f"Plugin installation failed: {output}")
+        return False
+
+    # If we get here, the package is not installed
+    return False
+
+
+def install_plugins_file():
+    """Install plugins from the plugins file"""
+
+    logger.info("Installing plugins from plugins file")
+
+    pf = settings.PLUGIN_FILE
+
+    if not os.path.exists(pf):
+        logger.warning(f"Plugin file {pf} does not exist")
+        return
+
+    pf = os.path.abspath(pf)
+
+    try:
+        pip_command('install', '-U', '-r', pf)
+    except subprocess.CalledProcessError as error:
+        output = error.output.decode('utf-8')
+        logger.error(f"Plugin installation failed: {output}")
+        return False
+    except Exception as exc:
+        logger.error(f"Plugin installation failed: {str(exc)}")
```

Comment:
Suggestion: The error message should probably indicate that the problem occurred during plugin file installation

Reply:
Fair point, have updated the messges now